### PR TITLE
Redesign + use components for settings

### PR DESCRIPTION
**Pull request by ashilkn**
This is an intermediate PR on the redesign of the settings screen so there will be unimplemented changes when compared with the Figma design and also at many places there are functions which return widgets which can be turned into stateless/stateful widgets which will be done towards the end of the redesign + use of components process.

https://user-images.githubusercontent.com/77285023/194210959-31b58431-a992-4062-826e-fb63ad7f6abd.mp4

Some different states of MenuItemWidget, CaptionedTextWidget and ToggleSwitchWidget components used together :  

Simulator Screen Shot - iPhone 12 mini - 2022-10-06 at 09 42 11
Simulator Screen Shot - iPhone 12 mini - 2022-10-06 at 09 46 51
Simulator Screen Shot - iPhone 12 mini - 2022-10-06 at 09 43 13
Simulator Screen Shot - iPhone 12 mini - 2022-10-06 at 09 49 21
Simulator Screen Shot - iPhone 12 mini - 2022-10-06 at 09 52 37
Simulator Screen Shot - iPhone 12 mini - 2022-10-06 at 09 53 21
Simulator Screen Shot - iPhone 12 mini - 2022-10-06 at 10 02 20
Simulator Screen Shot - iPhone 12 mini - 2022-10-06 at 10 04 13

### CODE REVIEW

File: lib/ui/components/captioned_text_widget.dart [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// leading icon can be passed without specifing size, this component sets size to 20x20
```
Comment (ua741) [inline lib/ui/components/captioned_text_widget.dart:1]: review the comment..? Also, this doesn't seems to be the right place to put it :)

File: lib/ui/components/captioned_text_widget.dart [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    this.subTitle,
+    this.textStyle,
+    this.makeTextBold = false,
+    this.textColor,
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final enteTheme = Theme.of(context).colorScheme.enteTheme;
```
Comment (ua741) [inline lib/ui/components/captioned_text_widget.dart:22]: I remember suggesting this pattern.. But may be we can create two vars here to reduce the verbosity in the remaining code..
```
final enteTextStyle
final enteColorSceme
```

Edit: After adding above comment, just discovered you are already similar stuff :)

File: lib/ui/components/captioned_text_widget.dart [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                      text: title,
+                    ),
+                    subTitle != null
+                        ? TextSpan(
+                            text: ' \u2022 ',
+                            style: enteTheme.textTheme.small.copyWith(
+                              color: enteTheme.colorScheme.textMuted,
+                            ),
+                          )
+                        : const TextSpan(text: ''),
```
Comment (ua741) [inline lib/ui/components/captioned_text_widget.dart:49]: Do we need this? May be in line 52 we can use
text: ' \u2022 subTitle'..,

File: lib/ui/settings/about_section_widget.dart [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        sectionOptionSpacing,
+        const AboutMenuItemWidget(
+          title: "Terms",
+          webPageTitle: "terms",
+          url: "https://ente.io/terms",
+        ),
+        sectionOptionSpacing,
+        const AboutMenuItemWidget(
+          title: "Privacy",
+          webPageTitle: "privacy",
```
Comment (ua741) [inline lib/ui/settings/about_section_widget.dart:45]: Now that text casing has changed on web too, we can use "Privacy" and "Terms" as title.

Comment (ua741) [reply lib/ui/settings/about_section_widget.dart:45]: After that, may be we can make `webPageTitle` as optional param. We can use `title` if `webPageTitle` is missing

File: lib/ui/settings_page.dart [hunk 5]
```diff
@@ -118,10 +111,10 @@ class SettingsPage extends StatelessWidget {
 
     return SingleChildScrollView(
       child: Padding(
-        padding: const EdgeInsets.symmetric(vertical: 12.0, horizontal: 20),
+        padding: const EdgeInsets.fromLTRB(16, 16, 16, 24),
         child: Center(
           child: ConstrainedBox(
-            constraints: const BoxConstraints(maxWidth: 350),
+            constraints: const BoxConstraints(maxWidth: 428),
```
Comment (ua741) [inline lib/ui/settings_page.dart:117]: Can we please test this on Simulator for iPad and SE, if not already done.

**Review by ua741 — APPROVED** [review]

File: lib/ui/components/captioned_text_widget.dart (hunk 1 above)
Comment (ashilkn) [reply lib/ui/components/captioned_text_widget.dart:1]: Oh yes, have fixed this

File: lib/ui/settings_page.dart (hunk 5 above)
Comment (ashilkn) [reply lib/ui/settings_page.dart:117]: Did test it on an iPad, tested on SE now and looks fine on both of them.

**Review by mnvr — APPROVED** [review]

**Pull request merged**